presto-spark-base/src/main/java/com/facebook/presto/spark/PhysicalResourceSettings.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class PhysicalResourceSettings
+{
+    public static final PhysicalResourceSettings DISABLED_PHYSICAL_RESOURCE_SETTING = new PhysicalResourceSettings(0, 0);
```

Comment:
I feel this could be an even better interface to prevent assigning 0s to the class.
```java
public static final PhysicalResourceSettings DISABLED_PHYSICAL_RESOURCE_SETTING = new PhysicalResourceSettings(0, 0, () -> {});

private final int executorCount;
private final int hashPartitionCount;

public PhysicalResourceSettings(int executorCount, int hashPartitionCount)
{
    this(
            executorCount,
            hashPartitionCount,
            () -> checkArgument(executorCount > 0 && hashPartitionCount > 0, "executorCount and hashPartitionCount should be positive"));
}

private PhysicalResourceSettings(int executorCount, int hashPartitionCount, Runnable check)
{
    check.run();
    this.executorCount = executorCount;
    this.hashPartitionCount = hashPartitionCount;
}